src/Haskell/Ide/Engine/Plugin/ApplyRefact.hs

```diff
@@ -106,6 +122,103 @@ applyAllCmd uri = pluginGetFile "applyAll: " uri $ \fp -> do
 
 -- ---------------------------------------------------------------------
 
+newtype HlintDataDir = HlintDataDir
+  { getHlintDataDir :: Maybe FilePath
+  }
+  deriving (Show, Eq, Read, Ord)
+
+
+instance ExtensionClass HlintDataDir where
+  initialValue = HlintDataDir Nothing
+
+data InitialisationError
+  = NoDataFiles FilePath
+  | InvalidDataDir FilePath
+  | Other
+
+ppInitialisationError :: InitialisationError -> String
+ppInitialisationError (NoDataFiles fp) =
+  "We could not find the data-files at: \"" <> fp <> "\""
+ppInitialisationError (InvalidDataDir fp) =
+  "The location \""
+    <> fp
+    <> "\" does not exist. We obtained that location from the environment variable: "
+    <> hlintDataDirEnvironmentVariable
+ppInitialisationError Other = "We could not determine why it failed."
+
+data Tool = Stack | Cabal
+
+showTool :: Tool -> String
+showTool Stack = "stack"
+showTool Cabal = "cabal"
+
+hlintName :: String
+hlintName = "hlint-" ++ VERSION_hlint
+
+hlintDataDirDefaultLocation :: MonadIO m => m FilePath
+hlintDataDirDefaultLocation = liftIO $ getXdgDirectory XdgData ("hie" </> "hlint")
+
+hlintDataDirEnvironmentVariable :: String
+hlintDataDirEnvironmentVariable = "HIE_HLINT_DATADIR"
```

Comment:
[Cabal already provides an environment variable for package data directories](https://www.haskell.org/cabal/users-guide/developing-packages.html?highlight=paths#accessing-data-files-from-package-code), and from whenever I last tested this it seemed to work with dependencies. Is there any possibility we could take advantage of `HASKELL_IDE_ENGINE_DATADIR` or `HLINT_DATADIR`?

Reply:
Ok, I thought that these were not honoured in dependencies.